dashboard/app/models/script.rb

```diff
@@ -342,6 +342,31 @@ def hoc_finish_url
     end
   end
 
+  def summarize
+    summary = {
+      id: id,
+      name: name,
+      stages: []
+    }
+    if trophies
+      summary[:trophies] = Concept.cached.map do |concept|
+        {
+          id: concept.name,
+          name: I18n.t("data.concept.description.#{concept.name}"),
+          bronze: Trophy::BRONZE_THRESHOLD,
+          silver: Trophy::SILVER_THRESHOLD,
+          gold: Trophy::GOLD_THRESHOLD
+        }
+      end
+    end
+
+    stages.select{|s| s.script_levels.to_a.count > 0}.sort_by(&:position).each do |stage|
```

Comment:
do we have stages with no levels? if so it might be causing problems elsewhere...
and I think the stages should be in order by position already